service/src/main/webapp/header.jsp

```diff
@@ -1,12 +1,28 @@
 <%@ page language="java" contentType="text/html; charset=UTF-8"
-    pageEncoding="UTF-8" import="java.util.*"%>
+    pageEncoding="UTF-8"
+    import="com.fecklessweasel.service.objectmodel.UserSession,com.fecklessweasel.service.UserSessionUtil" %>
 
 <div class="header">
     <div class="column-center">
         <input type="text" class="text-large" id="college" name="college" placeholder="Enter your university">
         <input type="text" class="text-large" id="class" name="class" placeholder="Enter your class (i.e. EECS-390)">
         <button type="submit" class="button-small" id="search-submit" name="search-submit">Go</button>
-        <a href="/account_create.jsp">Create an account</a>
+
+        <%-- Instantiate the session if there is one. You can use this in pages that include this. --%>
+        <%
+            UserSession authSession = UserSessionUtil.resumeSession(request);
+         %>
+
+        <%-- Show signup and login or logout button --%>
+        <%= authSession != null ? String.format(
+            "<form class='form-default' action='/servlet/user_session' method='post'>" +
+            "<input type='hidden' name='action' value='delete'/><input type='submit' " +
+            "class='button-large' id='submit' name='submit' value='Logout %s'/></form>",
+                authSession.getUser().getUsername()) :
+            "<a href='/account/create.jsp'>Sign up</a>" +
+            " <a href='/account/login.jsp'>Login</a>"
```

Comment:
Space is necessary to have a space between the Signup and Login links. Otherwise they appear as one big string. 

Should I move it to the previous line?